File: transpilers/other language/Python_11l_C/l11/_11l_to_cpp/tests/python_to_cpp/Rosetta Code/prime.cpp
#include "C:\!!BITBUCKET\11l-lang\_11l_to_cpp\11l.hpp"

auto k = 1000000;
auto n = k * 17;
auto primes = create_array({true}) * n;

struct CodeBlock1
{
    CodeBlock1()
    {
        _set<0>(primes, _set<1>(primes, false));

        for (auto i : range_ee(2, to_int(sqrt(n)))) {
            if (!primes[i])
                continue;
            for (auto j : range_el(i * i, n).step(i))
                primes.set(j, false);
        }

        for (auto i : range_el(0, n))
            if (primes[i]) {
                if (k == 1) {
                    print(i);
                    break;
                }
                k--;
            }
    }
} code_block_1;

int main()
{
}

File: transpilers/other language/Python_11l_C/l11/_11l_to_cpp/tests/python_to_cpp/Rosetta Code/prime.py
import math

#print([4] == [4])

#def testf():
k = 1000000
n = k * 17
primes = [True] * n
primes[0] = primes[1] = False

for i in range(2, int(math.sqrt(n)) + 1):
    if not primes[i]:
        continue
    for j in range(i * i, n, i):
        primes[j] = False

for i in range(n):
    if primes[i]:
        if k == 1:
            print(i)
            break
        k -= 1

#testf()